Applications/ConsoleReferenceSubscriber/Program.cs

```diff
@@ -44,6 +45,10 @@ public static class Program
         public const ushort NamespaceIndexSimple = 2;
         public const ushort NamespaceIndexAllTypes = 3;
 
+        // constant DateTime that represents the initial time when the metadata for the configuration was created
+        private static DateTime kTimeOfConfiguration = new DateTime(2021, 5, 1, 0, 0, 0, DateTimeKind.Utc);
+        private const string kDisplaySeparator = "------------------------------------------------";
+
         private static object m_lock = new object();
 
         public static void Main(string[] args)
```

Comment:
general note to the sample code to improve for FUTURE: The construction of the PubSub configuration looks complicated and error prone, it would be nice to have 'fluent API' or helper extension functions to build the structure content. Its unclear to me how some of the offset are calculated, so if this could be done by such an API would make the job a lot easier.

Reply:
Totally agree with your above observations and we are thinking of implementing such a "fluent API".